bench_method.go

```diff
@@ -38,7 +38,24 @@ type peerTransport struct {
 // benchmarkCaller exposes method to dispatch requests for benchmark.
 type benchmarkCaller interface {
 	// Call dispatches a request using the provided transport.
-	Call(transport.Transport) (time.Duration, error)
+	Call(transport.Transport) (benchmarkCallResult, error)
+}
+
+type benchmarkCallResult interface {
```

Comment:
We could call that `benchmarkResulter` ? What do you think? No need of putting `Call` in the name IMO.

Also, is there a purpose to have an interface ?

Reply:
It's the result of a single benchmark call, IMO`benchmarkResulter` wouldn't be the right name since this is not the result of entire benchmark rather of a single benchmark call. 

Currently, benchmark call returns just the Latency which is quite limiting for streaming benchmarks. #325 adds interface to return number of stream messages sent and received in streaming benchmarks

Reply:
Maybe there is use case in the future, but at least from this diff's scope, I don't see the value of having this interface. `benchmarkCallLatencyResult` seems to be enough for the use case of this change.